src/middlewared/middlewared/plugins/kubernetes_linux/restore.py

```diff
@@ -221,7 +222,25 @@ def restore_backup(self, job, backup_name, options):
                     'Failed to restore PVC(s) for %r chart release:\n%s', chart_release, '\n'.join(failed_pv_restores)
                 )
 
-            update_jobs.append(self.middleware.call_sync('chart.release.update', chart_release, {'values': {}}))
+            release = chart_releases_mapping[chart_release]
+            chart_path = os.path.join(release['path'], 'charts', release['chart_metadata']['version'])
+            # We will create any relevant crds now if applicable
+            crds_path = os.path.join(chart_path, 'crds')
+            failed_crds = []
+            if os.path.exists(crds_path):
+                for crd_file in os.listdir(crds_path):
+                    crd_path = os.path.join(crds_path, crd_file)
+                    try:
+                        self.middleware.call_sync('k8s.cluster.apply_yaml_file', crd_path)
+                    except Exception as e:
+                        failed_crds.append(f'Unable to create CRD(s) at {crd_path!r}: {e}')
+
+            if failed_crds:
+                self.logger.error(
+                    'Failed to restore CRD(s) for %r chart release:\n%s', chart_release, '\n'.join(failed_crds)
+                )
```

Comment:
Looks like a good fix for the CRD issue, although in the future we might want to standardise something like this in a more DRY fasion for both Update and Restore